_test_unstructured_client/integration/test_decorators.py

```diff
@@ -275,3 +284,69 @@ def test_integration_split_pdf_strict_mode(
         ],
     )
     assert len(diff) == 0
+
+
+def test_split_pdf_requests_do_retry(monkeypatch):
+    """
+    Test that when we split a pdf, the split requests will honor retryable errors.
+    """
+    # This is in a list so we can pass a reference to the mock function
+    number_of_502s = [1]
```

Comment:
nit: using `nonlocal` in `mock_send` instead of list would be more elegant and clear ;)
https://www.w3schools.com/python/ref_keyword_nonlocal.asp

Reply:
Perfect, thanks!